src/mixins/createFormFieldMixin.ts

```diff
@@ -77,66 +71,73 @@ export interface FormField<V> {
 	value?: string;
 }
 
-export type FormFieldMixin<V, S extends FormFieldMixinState<V>> = FormField<V> & Stateful<S>;
+export interface FormFieldOverride<V> {
+	/**
+	 * Add listener for a `valuechange` event, emitted when the value on the widget changes
+	 */
+	on(type: 'valuechange', listener: EventedListener<ValueChangeEvent<V>>): Handle;
+	on(type: string, listener: EventedListener<TargettedEventObject>): Handle;
+}
+
+export type FormFieldMixin<V, S extends FormFieldMixinState<V>> = FormField<V> & Stateful<S> & FormFieldOverride<V>;
 
 export interface FormMixinFactory extends ComposeFactory<FormFieldMixin<any, FormFieldMixinState<any>>, FormFieldMixinOptions<any, FormFieldMixinState<any>>> {
 	<V>(options?: FormFieldMixinOptions<V, FormFieldMixinState<V>>): FormFieldMixin<V, FormFieldMixinState<V>>;
 }
 
-const createFormMixin: FormMixinFactory = compose({
-		get value(this: FormFieldMixin<any, FormFieldMixinState<any>>): string {
-			return valueToString(this.state.value);
-		},
-
-		set value(this: FormFieldMixin<any, FormFieldMixinState<any>>, value: string) {
-			if (value !== this.state.value) {
-				const event = assign(createCancelableEvent({
-					type: 'valuechange',
-					target: this
-				}), {
-					oldValue: valueToString(this.state.value),
-					value
-				});
-				this.emit(event);
-				if (!event.defaultPrevented) {
-					this.setState({ value: stringToValue(event.value) });
-				}
-			}
-		},
-
-		nodeAttributes: [
-			function (this: FormFieldMixin<any, FormFieldMixinState<any>>): VNodeProperties {
-				const props: VNodeProperties = {};
-
-				if (this.type) {
-					props['type'] = this.type;
-				}
-				/* value should always be copied */
-				props.value = this.value;
-				if (this.state && this.state.name) {
-					props.name = this.state.name;
+const createFormMixin: FormMixinFactory = createStateful
+	.mixin({
+		mixin: <FormField<any>> {
+			get value(this: FormFieldMixin<any, FormFieldMixinState<any>>): string {
+				return valueToString(this.state.value);
+			},
+
+			set value(this: FormFieldMixin<any, FormFieldMixinState<any>>, value: string) {
+				if (value !== this.state.value) {
+					const event = assign(createCancelableEvent({
+						type: 'valuechange',
+						target: this
+					}), {
+						oldValue: valueToString(this.state.value),
+						value
+					});
+					this.emit(event);
+					if (!event.defaultPrevented) {
+						this.setState({ value: stringToValue(event.value) });
+					}
 				}
-				if (this.state.disabled) {
-					props['disabled'] = 'disabled';
+			},
+
+			nodeAttributes: [
+				function (this: FormFieldMixin<any, FormFieldMixinState<any>>): VNodeProperties {
+					const props: VNodeProperties = {};
+
+					if (this.type) {
+						props['type'] = this.type;
+					}
+					/* value should always be copied */
+					props.value = this.value;
+					if (this.state && this.state.name) {
+						props.name = this.state.name;
+					}
+					if (this.state.disabled) {
```

Comment:
@novemberborn this has already been fixed and merged into master.... if you revert to before this commit and pull the latest then it "should" work as expected.